### PR TITLE
Replace sys.exit() with raise SystemExit

### PR DESCRIPTION
**Pull request by janosh**
Reasoning explained in https://youtu.be/ZbeSPc5wL0g.

Split-out PR following https://github.com/kynan/nbstripout/pull/169#discussion_r985213917.

### CODE REVIEW

File: nbstripout/_nbstripout.py [hunk 1]
```diff
@@ -224,10 +224,10 @@ def install(git_config, install_location=INSTALL_LOCATION_LOCAL, attrfile=None):
         attrfile = _get_attrfile(git_config, install_location, attrfile)
     except FileNotFoundError:
         print('Installation failed: git is not on path!', file=sys.stderr)
-        sys.exit(1)
+        raise SystemExit(1)
```
Comment (kynan) [inline nbstripout/_nbstripout.py:227]:
```suggestion
        return 1
```

File: nbstripout/_nbstripout.py [hunk 2]
```diff
@@ -224,10 +224,10 @@ def install(git_config, install_location=INSTALL_LOCATION_LOCAL, attrfile=None):
...
     except CalledProcessError:
         print('Installation failed: not a git repository!', file=sys.stderr)
-        sys.exit(1)
+        raise SystemExit(1)
```
Comment (kynan) [inline nbstripout/_nbstripout.py:230]:
```suggestion
        return 1
```

File: nbstripout/_nbstripout.py [hunk 3]
```diff
@@ -262,7 +262,7 @@ def install(git_config, install_location=INSTALL_LOCATION_LOCAL, attrfile=None):
         if install_location == INSTALL_LOCATION_GLOBAL:
             print('Did you forget to sudo?', file=sys.stderr)
 
-        sys.exit(1)
+        raise SystemExit(1)
```
Comment (kynan) [inline nbstripout/_nbstripout.py:265]:
```suggestion
        return 1
```

File: nbstripout/_nbstripout.py [hunk 4]
```diff
@@ -274,10 +274,10 @@ def uninstall(git_config, install_location=INSTALL_LOCATION_LOCAL, attrfile=None
         attrfile = _get_attrfile(git_config, install_location, attrfile)
     except FileNotFoundError:
         print('Uninstall failed: git is not on path!', file=sys.stderr)
-        sys.exit(1)
+        raise SystemExit(1)
```
Comment (kynan) [inline nbstripout/_nbstripout.py:277]:
```suggestion
        return 1
```

File: nbstripout/_nbstripout.py [hunk 5]
```diff
@@ -274,10 +274,10 @@ def uninstall(git_config, install_location=INSTALL_LOCATION_LOCAL, attrfile=None
...
     except CalledProcessError:
         print('Uninstall failed: not a git repository!', file=sys.stderr)
-        sys.exit(1)
+        raise SystemExit(1)
```
Comment (kynan) [inline nbstripout/_nbstripout.py:280]:
```suggestion
        return 1
```

**Review by kynan — CHANGES_REQUESTED** [review]
Realised that we shouldn't be exiting from within functions but simply return.

**Comment by janosh** [comment]
> Realised that we shouldn't be exiting from within functions but simply return.

Was thinking about raising that point too.

**Review by kynan — APPROVED** [review]

**Pull request merged**